lessons/taquito/04/04.mdx

```diff
@@ -0,0 +1,160 @@
+        FAUCET_KEY.mnemonic.join(' '),
+        FAUCET_KEY.secret
+    );
+    
+    Tezos.setProvider({ rpc: 'https://carthagenet.SmartPy.io' });
+    
+    // Store the code here - 
+    // const CODE_JSON = 
+    
+    // Store the initial storage over here - 
+    // const STORAGE_JSON = 
+
+  answer: |
+    const FAUCET_KEY = {
+        "mnemonic": [
+            "chimney",
+            "item",
+            "course",
+            "pair",
+            "opinion",
+            "mass",
+            "receive",
+            "wrong",
+            "hurry",
+            "nest",
+            "oil",
+            "acoustic",
+            "section",
+            "salad",
+            "electric"
+        ],
+        "secret": "47184b1d8bf4dd550256817c3507cc4ccf112888",
+        "amount": "903205016",
+        "pkh": "tz1Q654L2qnt5mDM1Lbi4rMSTtnE5JHvD8M7",
+        "password": "9U5VOADM6l",
+        "email": "ryqwtuyj.ciwjnvdt@tezos.example.org"
+    }
+        
+    importKey(
+        Tezos,
+        FAUCET_KEY.email,
+        FAUCET_KEY.password,
+        FAUCET_KEY.mnemonic.join(' '),
+        FAUCET_KEY.secret
+    );
+    
+    Tezos.setProvider({ rpc: 'https://carthagenet.SmartPy.io' });
+    
+    // Store the code here - 
+    // const CODE_JSON = 
+    
+    // Store the initial storage over here - 
+    // const STORAGE_JSON = 
+    
+---
+### Intro
+Michelson is the language than runs on the Tezos network, rest all higher-level languages are an abstraction over it which directly compile to michelson.
+
+### Study time
+
+To deploy contracts through Taquito you need to grab the "Michelson JSON" and "Initial Storage JSON" for the contract that you wrote in SmartPy and this process is made SUPER EASY thanks to the [SmartPy IDE](smartpy.io/dev) - it generates "Storage JSON" and "Code JSON" for you which you can directly copy.
+
+### Show me an example
+If we've the following contract - 
+```python=
+import smartpy as sp
+
+class Counter(sp.Contract):
+    
+    def __init__(self, initialVal):
+        self.init(
+            value = initialVal
+        )
+    
+    def increment(self, amount):
+        self.data.value += amount
+        
+    
+    def decrement(self, amount):
+        self.data.value -= amount
+        
+        
+        
+@sp.add_test(name = "Counter")
+def test():
+    scenario = sp.test_scenario()
+    
+    counter = Counter(5)
+    scenario += counter
+```
+
+It generates the following **Code JSON** - 
+```json=
+[
+  { "prim": "storage", "args": [ { "prim": "int" } ] },
+  { "prim": "parameter", "args": [ { "prim": "unit" } ] },
+  { "prim": "code", "args": [ [ { "prim": "CDR" }, { "prim": "NIL", "args": [ { "prim": "operation" } ] }, { "prim": "PAIR" } ] ] }
+]
+```
+And Storage JSON - 
+```json=
+{ "int": "5" }
+```
+
+### #buidl-ing time
+ 
+#### New feature request!
+We're going to generate the JSON Code and Storage for the contract we want to originate and store it in our code.
+
+#### Step by step walkthrough
```

Comment:
This chapter also misses action for users to perform and get some output logged.

Reply:
It feels like it breaks in the middle. Mention that we will be deploying the contract in the coming chapters.

Reply:
Can't think of user action in this one yet.